### DISCUSSION TITLE
v2.0.0

### DISCUSSION THREAD

**Discussion by gucci-on-fleek** [gucci-on-fleek/lua-widow-control#24] [Announcements]
v2.0.0 of lua-widow-control has been released!

## Breaking Changes
- Page breaks may be slightly different
- Removed `\lwcemergencystretch` and `\lwcdisablecmd` in LaTeX. Please use the new key–value interface

## Other Changes
- Use expl3 for the LaTeX files (#20)
- Use a key–value interface for configuration with LaTeX (#11)
- Silence some extraneous `luatexbase` info messages
- Add a "debug mode" to print extra information (#12)
- Fix error message line wrapping
- Don't reset `\interlinepenalty` and `\brokenpenalty`
- Set and analyze `\displaywidowpenalty`
- Keep section headings with moved orphans (#17)
- Add the ability to configure the maximum paragraph cost (#22)
- Add a "strict" mode
- Use an improved cost function to select the best paragraph to lengthen (#23)
- Dozens of bug fixes
- Miscellaneous documentation updates

You can view the [full changelog on GitHub](https://github.com/gucci-on-fleek/lua-widow-control/compare/release-2c7201854d89535ef7c02f6c38486205677f1aa1...release-cea06ddad8dfcf15fa9ba2a86c6640648b9df523). 

Thanks to everyone for all your help!

**Comment by SarpC17** [comment]
Hey mate how can I get in touch with you asap?
